docs/proposals/templates-v2.md

```diff
@@ -0,0 +1,334 @@
+
+In order to template the filepath, there needs to be a place where metadata like this can be specified.
+I think this metadata should be stored alongside the templates themselves. That way, it can make use of all the same template variable available to the templates themselves.
+
+Conceptually, adding metadata to the templates is similar to Markdown frontmatter, though the choice of exact syntax for adding this metadata will have to be done with care since the templates can contain arbitrary contents including frontmatter.
+
+#### Example
+
+A workable syntax is still to be determined.
+While this syntax probably doesn't work as a solution, for this example I will demonstrate the concept using a second frontmatter block:
+
+```markdown
+<!-- The below front-matter block is for foam-specific template settings -->
+---
+<!-- The default filepath to use when using this template -->
+<!-- Relative paths are relative to the workspace, absolute paths are absolute -->
+<!-- Note that you can include VSCode snippet variables to template the path -->
+filepath: `journal/${CURRENT_YEAR}-${CURRENT_MONTH}-${CURRENT_DATE}_#{title}.md`
+---
+
+<!-- The actual contents of the template begin after the `---` thematic break immediately below this line-->
+---
+---
+created: ${CURRENT_YEAR}-${CURRENT_MONTH}-${CURRENT_DATE}T${CURRENT_HOUR}:${CURRENT_MINUTE}:${CURRENT_SECOND}
+tags: []
+---
+
+# ${title}
+```
+
+In this example, using this template improves the UX:
+
+In `Foam: Create New Note` workflow, having `foam.defaultNoteTemplate` point to a template with the `filepath` metadata allows for control over the filepath of without having to introduce any more UX steps to create a new note. It's still just a hotkey away and a title.
+
+As we'll see, when it comes to allowing daily notes to be templated, we don't even need to use `${title}` in our template, in which case we don't we don't even need to prompt for a title.
+
+In the `Create New Note From Template` workflow, during the step where we allow the user to customize the filepath, it will already templated according to the `filepath` in the template's metadata. This means that the user has to make fewer changes to the path, especially in cases where they want to include things like datetimes in the filenames. This makes it faster (eg. don't have to remember what day it is, and don't have to type it) and less error-prone (eg. when they accidentally type the wrong date).
+
+### Add support for daily note templates
+
+With the above features implemented, making daily notes use templates is simple.
+
+We create a new `foam.defaultDailyNoteTemplate` setting which is the template that `Foam: Open Daily Note` will always use (defaults to `null`)
+
+Both `Foam: Open Daily Note` and `Foam: Create New Note` can share all of the implementation code, with the only differences being the hotkeys used and the default template setting used (`foam.defaultDailyNoteTemplate` vs. `foam.defaultNoteTemplate`).
```

Comment:
I like the idea of hotkeys, we might want to pull this out in its own section?

Reply:
Hotkeys are great - but perhaps as an addition to the `foam-template` workspace-level settings rather than within the extension.

Current users of Foam might be frustrated if they're having to _unbind_ a hotkey vs binding one to "subscribe" to new functionality"

Reply:
yeah I agree - hotkeys are great, but they need to be done in the same paradigm that VS Code uses for all others keybindings, I don't think we should have a competing system there.
(i.e. we shouldn't have a `hotkey` key in the template or template configuration, the binding should live where all VS Code bindings live)

Reply:
Sounds like this is something we'll refactor/figure out as part of the implementation.

In the meantime, I've moved the section about user defined hotkeys to the "Extensions" section.